### ISSUE TITLE
Pure rust compression

### ISSUE THREAD

**Issue by marshallpierce** [jonhoo/hdrsample#64]
flate2 is fast, but it would be nice to have a default pure rust compression library so we wouldn't need to have the `serialization` feature stuff.

Presumably this would go along with allowing the user to specify which compression option to use, perhaps via generifying on the compressor.

**Comment by jonhoo** [comment]
Can't we just use https://github.com/sile/libflate ?

**Comment by marshallpierce** [comment]
Yep, almost certainly, though flate2 is faster by a fair bit. Just a simple matter of programming. :)
